anvil-python/anvil/commands/haproxy.py

```diff
@@ -100,34 +149,60 @@ def has_prompts(self) -> bool:
         return True
 
     def prompt(self, console: Console | None = None) -> None:
-        variables = questions.load_answers(
-            self.client, self._KEEPALIVED_CONFIG
-        )
+        variables = questions.load_answers(self.client, self._HAPROXY_CONFIG)
         variables.setdefault("virtual_ip", "")
+        variables.setdefault(
+            "haproxy_services_yaml", self._DEFAULT_SERVICES_CONFIG
+        )
 
         # Set defaults
         self.preseed.setdefault("virtual_ip", "")
+        self.preseed.setdefault(
+            "haproxy_services_yaml", self._DEFAULT_SERVICES_CONFIG
+        )
 
-        keepalived_config_bank = questions.QuestionBank(
-            questions=keepalived_questions(),
+        haproxy_config_bank = questions.QuestionBank(
+            questions=haproxy_questions(),
             console=console,
             preseed=self.preseed.get("haproxy"),
             previous_answers=variables,
             accept_defaults=self.accept_defaults,
         )
 
-        variables["virtual_ip"] = keepalived_config_bank.virtual_ip.ask()
+        variables["virtual_ip"] = haproxy_config_bank.virtual_ip.ask()
+
+        cert_filepath = haproxy_config_bank.ssl_cert.ask()
+        key_filepath = haproxy_config_bank.ssl_key.ask()
+        if cert_filepath is not None and key_filepath is not None:
+            with open(cert_filepath) as cert_file:
+                cert = cert_file.read()
+            with open(key_filepath) as key_file:
+                key = key_file.read()
+            if not os.path.isdir(HAPROXY_CERTS_DIR):
+                mkdir(HAPROXY_CERTS_DIR)
+            with open(
+                os.path.join(HAPROXY_CERTS_DIR, "haproxy.pem"), "w"
+            ) as combined_file:
+                combined_file.write(key + cert)
+            variables["haproxy_services_yaml"] = self._TLS_SERVICES_CONFIG
+            self.use_tls_termination = True
+        else:
+            LOG.debug(
+                "No certificate/key provided, skipping TLS configuration"
+            )
 
         LOG.debug(variables)
-        questions.write_answers(
-            self.client, self._KEEPALIVED_CONFIG, variables
-        )
+        questions.write_answers(self.client, self._HAPROXY_CONFIG, variables)
 
     def extra_tfvars(self) -> dict[str, Any]:
         variables: dict[str, Any] = questions.load_answers(
-            self.client, self._KEEPALIVED_CONFIG
+            self.client, self._HAPROXY_CONFIG
         )
-        variables["haproxy_port"] = 80
+        if self.use_tls_termination:
```

Comment:
Please add `variables["haproxy_services_yaml"] = self._TLS_SERVICES_CONFIG` here, after loading user answers from the variables

Reply:
In addition we could also remove ssl_cert, ssl_key from variables dict since Terraform does not need them